cli/tools/test/reporters/dot.rs

```diff
@@ -0,0 +1,483 @@
+// Copyright 2018-2023 the Deno authors. All rights reserved. MIT license.
+
+use super::fmt::format_test_error;
+use super::fmt::to_relative_path_or_remote_url;
+use super::*;
+
+pub struct DotTestReporter {
+  n: usize,
+  width: usize,
+  parallel: bool,
+  in_new_line: bool,
+  scope_test_id: Option<usize>,
+  cwd: Url,
+  child_results_buffer:
+    HashMap<usize, IndexMap<usize, (TestStepDescription, TestStepResult, u64)>>,
+  summary: TestSummary,
+}
+
+impl DotTestReporter {
```

Comment:
I still need to remove some of this code and deduplicate it against `PrettyTestReporter`. Will circle back to it tomorrow.

Reply:
I think this is good enough for now - from what I saw in the last week the whole testing module needs to be refactored and cleaned up. We can revisit in the future.